starfish/core/image/_filter/reduce.py

```diff
@@ -0,0 +1,123 @@
+import warnings
+from copy import deepcopy
+from typing import Callable, Iterable, Optional, Union
+
+import numpy as np
+
+from starfish.core.imagestack.imagestack import ImageStack
+from starfish.core.types import Axes
+from starfish.core.util import click
+from ._base import FilterAlgorithmBase
+
+
+class Reduce(FilterAlgorithmBase):
+    """
+    Reduces the dimensions of the ImageStack by applying a function
+    along one or more axes.
+
+    Parameters
+    ----------
+    dims : Axes
+        one or more Axes to project over
+    func : Union[str, Callable]
+        function to apply across the dimension(s) specified by dims.
+        If a functino is provided, it follow the form specified by
+        DataArray.reduce():
+        http://xarray.pydata.org/en/stable/generated/xarray.DataArray.reduce.html
+        The following strings are valid:
+            max: maximum intensity projection (applies numpy.amax)
+            mean: take the mean across the dim(s) (applies numpy.mean)
+            sum: sum across the dim(s) (applies numpy.sum)
+
+    See Also
+    --------
+    starfish.types.Axes
+
+    """
+
+    def __init__(
+        self, dims: Iterable[Union[Axes, str]], func: Union[str, Callable] = 'max'
+    ) -> None:
+
+        self.dims = dims
+
+        # If the user provided a string, convert to callable
+        if isinstance(func, str):
+            if func == 'max':
+                func = np.amax
+            elif func == 'mean':
+                func = np.mean
+            elif func == 'sum':
+                func = np.sum
+            else:
+                raise ValueError('func should be max, mean, or sum')
+        elif callable(func):
+            warnings.warn('User-specific functions are not logged')
```

Comment:
why is this true?

Reply:
Ah I didn't inspect the log very closely and I thought it wasn't actually storing the callable. I now see that it does, so I can remove the warning. Thanks!